cpp/examples/sde_seirvv.cpp

```diff
@@ -0,0 +1,98 @@
+    model.populations[{mio::sseirvv::InfectionState::RecoveredV1}] = 0;
+    model.populations[{mio::sseirvv::InfectionState::RecoveredV2}] = 0;
+    model.populations[{mio::sseirvv::InfectionState::ExposedV1V2}]  = 0;
+    model.populations[{mio::sseirvv::InfectionState::InfectedV1V2}]  = 0;
+    model.populations[{mio::sseirvv::InfectionState::RecoveredV1V2}] = 0;
+    model.populations[{mio::sseirvv::InfectionState::Susceptible}] =
+        total_population -
+        model.populations[{mio::sseirvv::InfectionState::ExposedV1}] -
+        model.populations[{mio::sseirvv::InfectionState::ExposedV2}] -
+        model.populations[{mio::sseirvv::InfectionState::InfectedV1}] -
+        model.populations[{mio::sseirvv::InfectionState::InfectedV2}] -
+        model.populations[{mio::sseirvv::InfectionState::RecoveredV1}] -
+        model.populations[{mio::sseirvv::InfectionState::RecoveredV2}] -
+        model.populations[{mio::sseirvv::InfectionState::ExposedV1V2}] -
+        model.populations[{mio::sseirvv::InfectionState::InfectedV1V2}] -
+        model.populations[{mio::sseirvv::InfectionState::RecoveredV1V2}];
+
+    model.parameters.get<mio::sseirvv::ContactPatterns>().get_baseline()(0, 0) = 1;
+    model.parameters.set<mio::sseirvv::TransmissionProbabilityOnContactV1>(0.076);
+    model.parameters.set<mio::sseirvv::TransmissionProbabilityOnContactV2>(0.076);
+    model.parameters.set<mio::sseirvv::TimeExposedV1>(5.33);
+    model.parameters.set<mio::sseirvv::TimeExposedV2>(5.33);      
+    model.parameters.set<mio::sseirvv::TimeInfectedV1>(17.2);
+    model.parameters.set<mio::sseirvv::TimeInfectedV2>(17.2 * 1.35);
+
+    model.check_constraints();
+
+    // The second variant enters with 100 individuals at some point tmid. The model does not fix
+    // the increase of totalpopulation caused by this injection.
+    auto sseirv = mio::sseirvv::simulate(t0, tmid, dt, model);
+    model.populations[{mio::sseirvv::InfectionState::Susceptible}] = sseirv.get_last_value()[0];
+    model.populations[{mio::sseirvv::InfectionState::ExposedV1}] = sseirv.get_last_value()[1];
+    model.populations[{mio::sseirvv::InfectionState::InfectedV1}]  = sseirv.get_last_value()[2];
+    model.populations[{mio::sseirvv::InfectionState::RecoveredV1}] = sseirv.get_last_value()[3];
+    model.populations[{mio::sseirvv::InfectionState::ExposedV2}]  = sseirv.get_last_value()[4];
+    model.populations[{mio::sseirvv::InfectionState::InfectedV2}]  = 100;
+    model.populations[{mio::sseirvv::InfectionState::RecoveredV2}] = sseirv.get_last_value()[6];
+    model.populations[{mio::sseirvv::InfectionState::ExposedV1V2}]  = sseirv.get_last_value()[7];
+    model.populations[{mio::sseirvv::InfectionState::InfectedV1V2}]  = sseirv.get_last_value()[8];
+    model.populations[{mio::sseirvv::InfectionState::RecoveredV1V2}] = sseirv.get_last_value()[9];
+    auto sseirv2 = mio::sseirvv::simulate(tmid, tmax, dt, model);
```

Comment:
I simplified the assignment and extended the comments.
```suggestion
    // Simulate the model up until tmid, with only the first variant.
    auto sseirv = mio::sseirvv::simulate(t0, tmid, dt, model);
    // Set the model population to the simulation result, so it is used as initial value for the second simulation.
    model.populations.array() = sseirv.get_last_value().cast<mio::UncertainValue<ScalarType>>();
    // The second variant enters with 100 individuals. This increases the model population to total_population + 100.
    model.populations[{mio::sseirvv::InfectionState::InfectedV2}] = 100;
    // Simulate the model up until tmid, with only the first variant.
    auto sseirv2 = mio::sseirvv::simulate(tmid, tmax, dt, model);
```
If you want to keep the elementwise assignment instead, please use `sseirv.get_last_value()[(size_t)mio::sseirvv::InfectionState::Susceptible]` etc. as RHS. Also, if you don't like that adding the 100 InfectedV2 increases the population size, you can (almost surely?) remove them from Susceptible.

Reply:
This looks much cleaner. The increase in population is consistent with the model Vincent from Jan's group is using so I am keeping it this way for now